Content.Trauma.Server/Chaplain/HolyFlammableSystem.cs

```diff
@@ -0,0 +1,503 @@
+        [Dependency] private readonly UseDelaySystem _useDelay = default!;
+        [Dependency] private readonly AudioSystem _audio = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly SpellbladeSystem _spellblade = default!; // Goobstation
+        [Dependency] private readonly IConfigurationManager _cfg = default!;
+
+        private EntityQuery<InventoryComponent> _inventoryQuery;
+        private EntityQuery<PhysicsComponent> _physicsQuery;
+
+        // This should probably be moved to the component, requires a rewrite, all fires tick at the same time
+        private const float UpdateTime = 1f;
+
+        private float _timer;
+
+        private readonly Dictionary<Entity<HolyFlammableComponent>, float> _fireEvents = new();
+
+        private int _addHeatFirestack = 1500;
+        public override void Initialize()
+        {
+            _inventoryQuery = GetEntityQuery<InventoryComponent>();
+            _physicsQuery = GetEntityQuery<PhysicsComponent>();
+
+            SubscribeLocalEvent<HolyFlammableComponent, MapInitEvent>(OnMapInit);
+            SubscribeLocalEvent<HolyFlammableComponent, StartCollideEvent>(OnCollide);
+            SubscribeLocalEvent<HolyFlammableComponent, TileFireEvent>(OnTileFire);
+            SubscribeLocalEvent<HolyFlammableComponent, RejuvenateEvent>(OnRejuvenate);
+            SubscribeLocalEvent<HolyFlammableComponent, ResistHolyFireAlertEvent>(OnResistFireAlert);
+            Subs.SubscribeWithRelay<HolyFlammableComponent, HolyExtinguishEvent>(OnExtinguishEvent);
+            Subs.SubscribeWithRelay<WeakToHolyComponent, HolyIgniteEvent>(OnHolyIgniteEvent);
+
+            SubscribeLocalEvent<HolyIgniteOnCollideComponent, StartCollideEvent>(HolyIgniteOnCollide);
+            //SubscribeLocalEvent<IgniteOnCollideComponent, LandEvent>(OnIgniteLand);
+            //SubscribeLocalEvent<IgniteOnCollideComponent, ProjectileHitEvent>(OnProjectileHit); // Goobstation
+
+            SubscribeLocalEvent<HolyIgniteOnMeleeHitComponent, MeleeHitEvent>(OnMeleeHit);
+
+            //SubscribeLocalEvent<ExtinguishOnInteractComponent, ActivateInWorldEvent>(OnExtinguishActivateInWorld);
+
+            SubscribeLocalEvent<IgniteOnHolyDamageComponent, DamageChangedEvent>(OnDamageChanged);
+        }
+
+        private void OnExtinguishEvent(Entity<HolyFlammableComponent> ent, ref HolyExtinguishEvent args)
+        {
+            // You know I'm really not sure if having AdjustFireStacks *after* Extinguish,
+            // but I'm just moving this code, not questioning it.
+            HolyExtinguish(ent, ent.Comp);
+            AdjustFireStacks(ent, args.FireStacksAdjustment, ent.Comp);
+        }
+
+        private void OnHolyIgniteEvent(Entity<WeakToHolyComponent> ent, ref HolyIgniteEvent args)
+        {
+            SetupEntity(ent);
+            EnsureComp<HolyFlammableComponent>(ent, out var flammable);
```

Comment:
`var flammable = EnsureComp<HolyFlammableComponent>(ent);`
you didnt use the bool return